environment.yml

```
@@ -31,84 +27,43 @@ channels:
   - conda-forge
 
 dependencies:
-  # Basics ------------------
+
   - python >= 3.8
   - pip
 
-  # Building NEST -----------
   - cmake >= 3.12
-  - cython
   - openmpi
   - boost >= 1.69
   - gsl >= 1.11
   - libtool
   - readline
   - hdf5
+  - cppcheck
+  - doxygen
+  - graphviz
 
-  # Testing NEST ------------
-  - pytest
-  - pytest-timeout
-  - pytest-xdist
-  - pytest-pylint
-  - pytest-mypy
-  - pytest-cov
-  - numpy
-  - pandas
-  - scipy
+  # mpi4py must be installed by conda in order to automatically find the path
+  # to OpenMPI installed by conda. Note that mpi4py also is included in
+  # requirements_pynest.txt in case someone wants to build an environment
+  # without conda. When this conda environment is built, the pip installation
+  # of mpi4py will be skipped since it is already installed by conda.
   - mpi4py
-  - data-science-types
-  - h5py
-  - terminaltables
-  - pycodestyle
-  - pydocstyle
-  - rstcheck
-  - mypy[version='>=0.8']
-  - cppcheck
+  
+  # assimulo is required to run doc/htmldoc/model_details/aeif_models_impementation.ipynb.
+  # The most convenient installation is via conda. This is, however, not a
+  # required package outside of the notebook detailing the aeif model implementation.
+  - assimulo
```

```suggestion
  #- assimulo
```
This is a dependency of an example, not of NEST itself. This could be kept here as a hint, but the clean way would be to remove it from NEST's requirements completely. (I have seen this being "solved" by `!pip install` lines in the notebook, but that has further disadvantages. If you want to put it into the notebook, please add it only as a comment `# !pip install` so that it doesn't break users environments, unless actively uncommented by the user.)

---

I removed `assimulo` in accordance with https://github.com/nest/nest-simulator/pull/2906.